system/Session/Session.php

```diff
@@ -178,22 +179,27 @@ public function __construct(SessionHandlerInterface $driver, App $config)
 		$this->sessionTimeToUpdate      = $config->sessionTimeToUpdate ?? $this->sessionTimeToUpdate;
 		$this->sessionRegenerateDestroy = $config->sessionRegenerateDestroy ?? $this->sessionRegenerateDestroy;
 
+		/**
+		 * @var CookieConfig
+		 */
+		$cookie = config('Cookie');
+
 		//---------------------------------------------------------------------
 		// DEPRECATED COOKIE MANAGEMENT
 		//---------------------------------------------------------------------
-		$this->cookieDomain   = $config->cookieDomain ?? $this->cookieDomain;
-		$this->cookiePath     = $config->cookiePath ?? $this->cookiePath;
-		$this->cookieSecure   = $config->cookieSecure ?? $this->cookieSecure;
-		$this->cookieSameSite = $config->cookieSameSite ?? $this->cookieSameSite;
+		$this->cookiePath     = $cookie->path ?? $config->cookiePath ?? $this->cookiePath;
+		$this->cookieDomain   = $cookie->domain ?? $config->cookieDomain ?? $this->cookieDomain;
+		$this->cookieSecure   = $cookie->secure ?? $config->cookieSecure ?? $this->cookieSecure;
+		$this->cookieSameSite = $cookie->samesite ?? $config->cookieSameSite ?? $this->cookieSameSite;
 
 		$this->cookie = Cookie::create($this->sessionCookieName, '', [
 			'expires'  => $this->sessionExpiration === 0 ? 0 : time() + $this->sessionExpiration,
-			'raw'      => $config->cookieRaw ?? false,
-			'domain'   => $config->cookieDomain ?? '',
-			'path'     => $config->cookiePath ?? '/',
-			'secure'   => $config->cookieSecure ?? false,
+			'path'     => $this->cookiePath,
+			'domain'   => $this->cookieDomain,
+			'secure'   => $this->cookieSecure,
 			'httponly' => true, // for security
-			'samesite' => $config->cookieSameSite ?? Cookie::SAMESITE_LAX,
+			'samesite' => $this->cookieSameSite,
+			'raw'      => $cookie->raw ?? false,
```

Comment:
Do not add dependence to the deprecated cookie properties. It is deprecated so that it will no longer be used in active code and can be safely removed in the future but still preserving for those users who depended on such. By deprecating but still actively depending on those properties would defeat the purpose of the deprecation.